addons/website/models/website_rewrite.py

```diff
@@ -105,6 +105,10 @@ def _check_url_to(self):
                 except ValueError as e:
                     raise ValidationError(_('"URL to" is invalid: %s') % e)
 
+    def _modify_url_to_if_needed(self, vals):
```

Comment:
Why do we need such check function?  If I am not wrong, we can do same in `def _check_url_to` too.(it already contains some URL validation condition, IMO we should move there.)  :face_with_diagonal_mouth:

Reply:
If we write the same code inside `def _check_url_to` method whenever we make changes to any parameter which is written inside `@api.constrains()` then `def _check_url_to` method again runs and it will run infinite time so we need to make a new function and we are calling that function from `create` and `write` method.